image-processing/main.py

```diff
@@ -20,36 +58,286 @@ def face_detection(uri):
         or returns an errors resonse in string format
     """
     vision_client = vision.ImageAnnotatorClient()
-    print(vision_client)
 
     image = vision.Image()
     image.source.image_uri = uri
 
     response = vision_client.face_detection(image=image)
+
     faceAnnotations = response.face_annotations
 
     # Making sure that there's only one person in the frame
     if len(faceAnnotations) != 1:
         return "Please ensure exactly ONE face is in the image."
 
-    # Lables of likelihood from google.cloud.vision.enums
+    face = faceAnnotations[0]
+
+    result = face_check(face)
+
+    if result != "Success":
+        return result
```

Comment:
catch exception here

Reply:
I think it's better to catch it in detect_and_process()